src/doc_builder/commands/commit.py

```diff
@@ -0,0 +1,145 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import base64
+import re
+from pathlib import Path
+
+import requests
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport
+from gql.transport.exceptions import TransportQueryError
+
+
+_re_head_oid = re.compile(r"\"sha\":\"([0-9a-z]+)\"")
+
+
+def get_head_oid(repo_id, token, branch="main"):
+    """
+    Returns last commit sha from repostory `repo_id` & branch `branch`
+    """
+    res = requests.get(
+        f"https://api.github.com/repos/{repo_id}/git/refs/heads/${branch}",
+        headers={"Authorization": f"bearer {token}"},
+    )
+    if res.status_code == 200:
+        head_oid = _re_head_oid.search(res.text)[1]
+        return head_oid
+    else:
+        raise Exception("get_head_oid failed: {{status: {res.status}, msg:{res.text}}}")
```

Comment:
why not parse the JSON btw?

Reply:
handled in https://github.com/huggingface/doc-builder/pull/214/commits/6ce101dfed4531ac19b405cdf86a3de5df44008d

I tend to over-use regex 😊

Reply:
definitely, @Pierrci's fault 😊

Reply:
mmmm I'm not that sure haha